### ISSUE TITLE
Can’t edit style Gantry Template for Joomla v4 from admin panel

### ISSUE THREAD

**Issue by Adminkg** [gantry/gantry5#3141]
I can’t edit the Gantry from admin panel template. When I click “Gantry - Default” “(domain)/administrator/index.php?option=com_gantry&task=template.edit&id=10” I get “404 Component not found”. The components are there. The website works. All other administrative functions work.

I don’t know where to look. The last thing I changed was what I added in the “Text resizer” in Gantry

Sorry for my English. It’s partly Google Translate.

Kris

**Comment by N8Solutions** [comment]
Hello @Adminkg A few things for you to try would be to:

- Reinstall the Gantry Framework and the template you are using, making sure both are the latest version. Don't worry, reinstalling them won't overwrite any custom settings.
- Try lowering the version of PHP that you are using. The latest version of the Gantry Framework is suppose to fix a lot of PHP 8 issues but it is always possible that your hosting setup may be different. Start by going to v7.4 first, then try v8 & v8.1.
- If you are using a CDN, make sure you have excluded the administrator section of your site from being cached by the CDN.

Hope this helps.

**Comment by Adminkg** [comment]
Yes, it can help. However, I don't want to do that because I have too outdated Joomla and Gantry. I will have to redo the page.

I thought someone might have had this happen and the diagnosis would be simple. I searched the internet but couldn't find a similar problem. Sure, someone had it, but probably didn't describe it.

Everything worked for years and nothing new was installed. PHP the same. Added/deleted articles, menu items, etc. and all of a sudden I can't edit the template. I checked the database, website code, file dates, compared with the website backup, deleted temporary files. All good. I don't know where to look.

Thanks for the tips
Regards
Kris

**Comment by Adminkg** [comment]
Versions
Joomla! 3.7.2 Stable [ Amani ] 22-May-2017 09:46 GMT
Gantry Starting Template for Joomla! v4.1.21

**Comment by Adminkg** [comment]
Found. Not mine but same message.
![Untitled](https://user-images.githubusercontent.com/130636852/231739724-d1d2b70d-62fd-4746-8029-2a0ce348c25b.jpg)

**Comment by Adminkg** [comment]
Are these settings ok?
![phpMyAdmin](https://user-images.githubusercontent.com/130636852/231742567-a0b95576-4e17-49e2-9818-21637db15e29.png)

**Comment by Adminkg** [comment]
Sorry. It is in Polish.
![Gantry](https://user-images.githubusercontent.com/130636852/231743684-b5edcb9a-e796-4d5b-b56f-bc06ffb5cb49.png)

**Comment by Adminkg** [comment]
gantry
Website Template 4.1.21 February 28, 2014 RocketTheme, LLC none 10004

Gantry
Site Component 4.1.24 April 12, 2014 RocketTheme, LLC none 10002

Gantry
Website Library 4.1.24 April 12, 2014 RocketTheme, LLC none 10031

System - Gantry
Site Addon 4.1.24 April 12, 2014 RocketTheme, LLC system 10003

**Comment by Adminkg** [comment]
I forgot the headline
**State, Name, Client, Type, Version, Date, Author, Directory Bundle ID, ID**

**Comment by N8Solutions** [comment]
@Adminkg OK, You are using Gantry 4. At the very least you should update Gantry 4 to the latest version which is v4.1.43.

The version you are using is very outdated. If you update to the latest version of Gantry 4 it may solve your problems.

You should also consider upgrading to Joomla 4 with Gantry 5. Gantry 5 has an Outline that you can use that has the module positions specific for Gantry 4. It will still require some tinkering to get it right but it won't be as big of a rebuild as you would expect.

Make a backup of your site, and start with updating Gantry 4. Then, when you are ready, you should update to Gantry 5, and make all the styling changes, then you can migrate to Joomla 4.

Joomla 3 will no longer be supported after Aug. 17th. Support for Gantry 4 has pretty much ended as well but it will be final at the same time as Joomla 3 because it will not be made compatible for Joomla 4. The highest version of PHP you can use with Gantry 4 is v7.4. Your site is already at a security risk due to it being so outdated and will be even more so if you keep it this way after Joomla 3 is end of life.

Another thing for you to do before updating to the latest version of Gantry 4 is to turn on error reporting so you can what is causing the error message. With such an old version, if you have added any new extensions or updated any of the existing extensions, this could cause a conflict creating the error message you are seeing.

**Comment by Adminkg** [comment]
Thanks for explaining the compatible versions. I wouldn't even try without it.
I will try all this on a copy in xamp. The site has a rather complicated menu and dozens of articles.
I just didn't install anything new. I only enabled "text resizing" in the Gantry template.

I still have a question, have you seen the screenshots from the mysql database in the post above? I'm not sure if it's okay there in jos3_extensions table?

Thanks a lot to you [N8Solutions](https://github.com/N8Solutions)

**Comment by Adminkg** [comment]
Funny. Someone in the world, however, had such a problem, but without a solution. Unfortunately.
[](https://rockettheme.com/forum/configuration-and-security/257885-solved-error-404-component-not-found-when-editing-gantry-template)

**Comment by N8Solutions** [comment]
> I still have a question, have you seen the screenshots from the mysql database in the post above? I'm not sure if it's okay there in jos3_extensions table in the field "com_gantry".
> In which file can I disable "text resizing"

@Adminkg  First things first, you should consider changing the table prefix from "jos3" to something else as that is the old default and makes it easy for a website like yours to be hacked. You can do this automatically by using a tool like "Admin Tools" from Akeeba Backup. Though, since you are using such an old website, you will need to search for an older version that is compatible to yours from their archives.

Now on to solving your issue.
In phpMyAdmin, search your database for the table `"jos3_template_styles"`
Once you have found that table, you will need to select the entry labeled `"gantry - Default"`, choose the option to edit the table.
in the `"params"` section, you need to search for `"fontsizer":{"enabled":"0","position":"feature-b"}`. That is the default. A Zero (0) means it is turned off. So if you turned it on, it should say `"enabled":"1"`, changing that to 0 and saving it will turn it off. If that is the problem then everything should start working again.

It goes without saying, but before you do anything like this, please make sure you take a backup of your website first! I'm not liable for any problems that occur from you making changes to your website this way.

Good luck and let me know if this helps.

**Comment by Adminkg** [comment]
Thank you for your answer.
One last question before I start.
Should the "Enabled" field of the com_gantry record be 0 or 1?
[com_gantry.pdf](https://github.com/gantry/gantry5/files/11229559/com_gantry.pdf)

**Comment by N8Solutions** [comment]
> Thank you for your answer. One last question before I start. Should the "Enabled" field of the com_gantry record be 0 or 1?

@Adminkg It should be 1

Let me know how it goes.

**Comment by N8Solutions** [comment]
You're very welcome! Best of luck to you @Adminkg

**Comment by N8Solutions** [comment]
I'm glad you got it to work with the information I gave you.

About PHP, Gantry 4 is not compatible with anything higher than PHP v7.4. If you try to use PHP v8.0, v8.1, or v8.2 it will fail and there's nothing you can do about it because it is simply not compatible.

The thing to remember is that all of the modules are assigned to positions that the Gantry 4 template uses. Rather than reinstalling, what you can do is what I mentioned before.

- Install Gantry 5 framework and free template like Hydrogen or Helium
- Make it the default template
- Uninstall Gantry 4 because leaving it installed will mess things up later
- Load the Gantry 4 Outline which will propagate the Outline with Module position names like what is used in Gantry 4
- You'll need to check the styling as some things will need to be replaced.

You should refer to the Gantry 5 docs fore more information.
https://docs.gantry.org/
If you need help, feel free to reply here.


kind regards,

Michael

**Comment by Adminkg** [comment]
I exercise. 

First I made a copy (Akeeba) from production server.
- Xampp PHP 5.6.
- Copy recovery
- Uninstall unnecessary or problematic components and uninstalling Gantry.
- Default template - standard.
- Joomla update to this version which updates itself (3.10.11)
- Installing a newer version of Akeeba. Copy.
- Xampp PHP 8.2
- Copy recovery (Akeeba - various minor issues, versions).
- Uninstall unnecessary or problematic components.
- Update Joomla via Joomla to 4.3. 
- Installing components, Gantry5 and templates.

It is described in short.

He's working! At least on a copy. Not everything, but the structure, menu, articles, etc. are.

Now I try customizing for Gantry5. I'm learning the documentation first. It's necessary to.

I'll try to do the whole thing on Xampp and transfer to the production server. It might work.

Thanks Michael
regards,
Kris

**Comment by jameswadsworth** [comment]
I came across this problem when trying to update an old webiste for a client: "Component not found" when trying to edit Gantry template. Gantry 4 has been installed on this website for many years but the Gantry 4 Templates Manager (the administrator com_gantry component) has ended up with state -1 in the #__extensions table meaning Joomla thinks the component is yet to be installed. The solutions are 2:
1) change directly in the database of state field for the com_gantry in the table #__extensions from -1 to 1
2) in the Joomla admin go Manage -> Extensions -> Discover, select the Gantry 4 Templates Manager and click on the install button
